internal/core/common_config/main.go

```diff
@@ -56,23 +66,62 @@ func Main(ctx context.Context, cancel context.CancelFunc) {
 			return lc
 		},
 	})
-
 	secretProvider, err := secret.NewSecretProvider(nil, environment.NewVariables(lc), ctx, startupTimer, dic, common.CoreCommonConfigServiceKey)
 	if err != nil {
 		lc.Errorf("failed to create Secret Provider: %v", err)
 		os.Exit(1)
 	}
 
 	lc.Info("Secret Provider created")
+	var accessToken string
+	var getAccessToken types.GetAccessTokenCallback
+
+	if secretProvider != nil {
+		getAccessToken = func() (string, error) {
+			accessToken, err = secretProvider.GetAccessToken("consul", common.CoreCommonConfigServiceKey)
+			if err != nil {
+				return "", fmt.Errorf("failed to get Configuration Provider access token: %s", err.Error())
+			}
+			return accessToken, err
+		}
+		_, err = getAccessToken()
+		if err != nil {
+			lc.Errorf("failed to get Access Token for config provider: %s", err.Error())
+			os.Exit(1)
+		}
+	}
```

Comment:
probably it's a good idea to add to handle else- case here. we don't wanna continue in this case....

Reply:
this if statement is not used anymore, so I think this is addressed.